viam-cartographer.go

```diff
@@ -346,11 +347,25 @@ func (cartoSvc *cartographerService) GetSLAMProcessConfig() pexec.ProcessConfig
 	args = append(args, "-port="+cartoSvc.port)
 	args = append(args, "--aix-auto-update")
 
+	target := cartoSvc.executableName
+
+	appDir := os.Getenv("APPDIR")
+
+	if appDir != "" {
+		// The carto grpc server is expected to be in
+		// /usr/bin/ if we are running in an AppImage
+		target = appDir + "/usr/bin/" + strings.TrimPrefix(target, "/")
+	}
+
 	return pexec.ProcessConfig{
-		ID:      "slam_cartographer",
-		Name:    cartoSvc.executableName,
-		Args:    args,
-		Log:     true,
+		ID:   "slam_cartographer",
+		Name: target,
+		Args: args,
+		Log:  true,
+		// In appimage this is set to the appimage
+		// squashfs mount location (/tmp/.mountXXXXX)
+		// Otherwise, it is an empty string
+		CWD:     os.Getenv("APPRUN_RUNTIME"),
```

Comment:
Could we confirm that even with this change, we can still use cartographer outside of an appimage, e.g. can still run cartographer on a mac?